projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -2,630 +2,339 @@ import { IgxGridNavigationService } from '../grid-navigation.service';
-                }
-            }
-        } else if (currentRowIndex !== 0) {
-            // handle scenario when prev item is child grid but is not yet in view
-            const isPrevChildGrid = this.getIsChildAtIndex(currentRowIndex - 1);
-            if (!isPrevChildGrid) {
-                super.navigateUp(rowElement, selectedNode);
-            } else {
-                this.scrollGrid(this.grid, -rowElement.offsetHeight,
-                    () => {
-                        rowElement = this.getRowByIndex(currentRowIndex);
-                        this.navigateUp(rowElement, selectedNode);
-                    });
+        if (this.activeNode && event.altKey) {
+            event.preventDefault();
+            const row = this.grid.getRowByIndex(this.activeNode.row) as IgxHierarchicalRowComponent;
+            if (row.added) {
+                return;
             }
-        } else if (this.grid.parent !== null &&
-            currentRowIndex === 0) {
-            // move to prev row in sibling layout or parent
-            this.focusPrev(visibleColumnIndex);
-        }
-    }
-    public navigateDown(rowElement, selectedNode: ISelectionNode) {
-        if (selectedNode.isSummaryRow) { return; }
-        const nextElem = rowElement.nextElementSibling;
-        const visibleColumnIndex = selectedNode.column;
-        const currentRowIndex = selectedNode.row;
-        if (nextElem) {
-            // next elem is in DOM
-            const nodeName = nextElem.children[0].nodeName.toLowerCase();
-            const isNextElemChildGrid = nodeName.toLowerCase() === 'igx-child-grid-row';
-            if (isNextElemChildGrid) {
-                this.focusNextChild(nextElem, visibleColumnIndex, this.grid);
-            } else {
-                if (this.grid.parent !== null) {
-                    // currently navigating in child grid
-                    this._navigateDownInChild(rowElement, currentRowIndex, visibleColumnIndex);
-                } else {
-                    super.navigateDown(rowElement, selectedNode);
-                }
+            const collapse = row.expanded && (key === 'left' || key === 'arrowleft' || key === 'up' || key === 'arrowup');
+            const expand = !row.expanded && (key === 'right' || key === 'arrowright' || key === 'down' || key === 'arrowdown');
+            if (collapse) {
+                this.grid.gridAPI.set_row_expansion_state(row.rowID, false, event);
+            } else if (expand) {
+                this.grid.gridAPI.set_row_expansion_state(row.rowID, true, event);
             }
-        } else if (currentRowIndex !== this.grid.dataView.length - 1) {
-            // scroll next in view
-            super.navigateDown(rowElement, selectedNode);
-        } else if (this.grid.parent !== null &&
-            currentRowIndex === this.grid.dataView.length - 1) {
-            // move to next row in sibling layout or in parent
-            this.focusNext(visibleColumnIndex);
+            return;
```

Comment:
this logic seems quite similar to the logic implemented in the handleAlt() method inside the base navigation service .. can you please check to see if it is possible to use it instead :)